client/ayon_houdini/plugins/publish/collect_usd_layers.py

```diff
@@ -70,6 +70,39 @@ def process(self, instance):
             return
 
         rop_node = hou.node(instance.data["instance_node"])
+        lop_path = rop_node.evalParm("loppath")
+        lop_node = hou.node(lop_path)
+
+        self.log.debug(f"ROP Node: {rop_node.path()} references LOP Node: {lop_node.path()}")
+
+        def find_geoclipsequence(node):
+            if node.type().name() == "geoclipsequence":
+                return node
+            for input_node in node.inputs():
+                if input_node:
+                    result = find_geoclipsequence(input_node)
+                    if result:
+                        return result
+            return None
+
+        geoclip_node = find_geoclipsequence(lop_node)
+        self.log.debug(f" geoclip node {geoclip_node}")
+
+        if geoclip_node:
+            self.log.debug("True geoclip node")
+            if geoclip_node.parm("saveclipfilepath"):
+                clip_path = geoclip_node.evalParm("saveclipfilepath")
+            else:
+                clip_path = None
+            manifest_path = geoclip_node.evalParm("manifestfile")
+            topology_path = geoclip_node.evalParm("topologyfile")
+
+            self.log.debug(f"Found GeoClipSequence Node: {geoclip_node.path()}")
+            self.log.debug(f"Clip Path: {clip_path}")
+            self.log.debug(f"Manifest Path: {manifest_path}")
+            self.log.debug(f"Topology Path: {topology_path}")
+        else: 
+            self.log.debug("False not a geoclipsequence node")
```

Comment:
Thanks for the contribution - even though this shows what _may need to be collected_ from a GeoClip this is not the approach we should be looking for unfortunately. This is _VERY SPECIFIC_ to the GeoClip node - and NOT to USD data in general. Instead we should be querying the USD scene and its state to know about the files.

The GeoClip node may not be there (it may have somehow been sublayered in, or come from an object merge from somewhere else, etc. etc. Or it could be inside a custom HDA.

We _should_ be querying the relevant files from the USD Stage or its Sdf Layers. It's something I can look into given priority and time - however so far that hasn't been there. :)